build/devenv/lib.go

```diff
@@ -0,0 +1,106 @@
+package ccv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-ccv/devenv/cciptestinterfaces"
+	"github.com/smartcontractkit/chainlink-ccv/devenv/evm"
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+)
+
+type Lib struct {
+	envOutFile string
+	cfg        *Cfg
+	l          *zerolog.Logger
+}
+
+func NewLib(logger *zerolog.Logger, envOutFile string) (*Lib, error) {
+	cfg, err := LoadOutput[Cfg](envOutFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load environment output: %w", err)
+	}
+
+	lib := &Lib{
+		envOutFile: envOutFile,
+		cfg:        cfg,
+		l:          logger,
+	}
+
+	if err := lib.verify(); err != nil {
+		return nil, fmt.Errorf("invalid library object: %w", err)
+	}
+
+	return lib, nil
+}
+
+// NewImpl is a convenience function that fetches a specific impl from the library.
+func NewImpl(logger *zerolog.Logger, envOutFile string, selector uint64) (cciptestinterfaces.CCIP17ProductConfiguration, error) {
```

Comment:
Should we consider accepting a context here? That way we could pass it to the lib. I think most callers have a context as well